shell/components/Inactivity.vue

```diff
@@ -0,0 +1,215 @@
+<script>
+import ModalWithCard from '@shell/components/ModalWithCard';
+import { Banner } from '@components/Banner';
+import PercentageBar from '@shell/components/PercentageBar.vue';
+import throttle from 'lodash/throttle';
+import { MANAGEMENT } from '@shell/config/types';
+import { DEFAULT_PERF_SETTING, SETTING } from '@shell/config/settings';
+
+export default {
+  name:       'Inactivity',
+  components: {
+    ModalWithCard, Banner, PercentageBar
+  },
+  data() {
+    return {
+      enabled:             null,
+      isOpen:              false,
+      isInactive:          false,
+      showModalAfter:      null,
+      inactivityTimeoutId: null,
+      courtesyTimer:       null,
+      courtesyTimerId:     null,
+      courtesyCountdown:   null,
+      trackInactivity:     throttle(this._trackInactivity, 1000),
+    };
+  },
+  async mounted() {
+    const settingsString = await this.$store.dispatch('management/find', { type: MANAGEMENT.SETTING, id: SETTING.UI_PERFORMANCE });
+
+    const settings = settingsString?.value ? JSON.parse(settingsString.value) : DEFAULT_PERF_SETTING;
+
+    if (!settings || !settings?.inactivity || !settings?.inactivity.enabled) {
+      return;
+    }
+
+    this.enabled = settings?.inactivity?.enabled || false;
+
+    const thresholdToMinutes = settings?.inactivity?.threshold * 60;
+
+    this.showModalAfter = parseInt(thresholdToMinutes * 0.9); // convert minutes, triggers at 90% completion
+    this.courtesyTimer = Math.floor(thresholdToMinutes * 0.1); // 10% of the threshold in minutes, goes by for the remaining 10% of the threshold
```

Comment:
Was this the end point decision on ratio? If this is high then that's a lot of time. For example 60 minute inactivity means 6 minutes of the modal on screen

Reply:
Any news on this?

Reply:
I've pushed a fix with this in